src/main.rs

```diff
@@ -37,10 +41,34 @@ use queue::Queue;
 const SNARE_CONF_SEARCH: &[&str] = &["/etc/snare.conf", "~/.snare.conf"];
 
 pub(crate) struct Snare {
+    /// The location of snare.conf; this file will be reloaded if SIGHUP is received.
+    conf_path: PathBuf,
+    /// The current configuration: note that this can change at any point due to SIGHUP.
     config: Mutex<Config>,
+    /// The current queue of incoming jobs.
     queue: Mutex<Queue>,
+    /// The read end of the pipe used by the httpserver and the SIGHUP handler to wake up the job
+    /// runner thread.
     event_read_fd: RawFd,
+    /// The write end of the pipe used by the httpserver and the SIGHUP handler to wake up the job
+    /// runner thread.
     event_write_fd: RawFd,
+    /// Has a SIGHUP event occurred? If so, the jobrunner will process it, and set this to false in
+    /// case future SIGHUP events are detected.
+    sighup_occurred: Arc<AtomicBool>,
+}
+
+impl Snare {
+    /// Check to see if we've received a SIGHUP since the last check. If so, we will reload the
+    /// config file. **Note that because snare has multiple threads, the config file can change at
+    /// any arbitrary point, not just after calling this function.**
+    fn check_for_hup(&self) {
+        if self.sighup_occurred.load(Ordering::Release) {
```

Comment:
Is the ordering supposed to be acquire here?

Reply:
I think `Relaxed` is fine here as we don't need any other read/writes to have occurred before/after this.

Reply:
But we have `Release`.

Perhaps both this and the ordering in a few lines time should both be relaxed?

(If the config weren't in a mutex, you'd certainly want acquire/release, otherwise another thread may see the config mid-move)

Reply:
You're totally right: they should both be `Relaxed` and one of the later commits (https://github.com/softdevteam/snare/pull/15/commits/ff45ea1e03b8c4d86a38cd6c04e38e691c88c736#diff-639fbc4ef05b315af92b4d836c31b023R66) fixes that. I don't know why I ever thougt `Release` was the correct ordering!